File: Data Structures & Algorithms/longest-consecutive-sequence/submission-3.java
class Solution {
    public int longestConsecutive(int[] nums) {
        Set<Integer> set = new HashSet<>();
        for (int n : nums) set.add(n);

        int longest = 0;

        for (int n : set) {
            // only start counting if n is the start of a sequence
            if (!set.contains(n - 1)) {
                int length = 1;
                int cur = n;

                while (set.contains(cur + 1)) {
                    cur++;
                    length++;
                }

                longest = Math.max(longest, length);
            }
        }

        return longest;
    }
}

File: Data Structures & Algorithms/house-robber/submission-3.py
class Solution:
    def rob(self, nums: List[int]) -> int:
        max_until_prev: int = 0 #       dp[i-1]
        max_until_prev_prev: int = 0 #  dp[i-2]

        for n in nums:
            current = max(max_until_prev, max_until_prev_prev + n) # dp [i]

            max_until_prev_prev = max_until_prev
            max_until_prev = current
        
        return max_until_prev